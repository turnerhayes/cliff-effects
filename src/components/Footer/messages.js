export default {
  header: { text: { id: 'footer.header.text' }},

  cfbCredit: { id: 'footer.cfbCredit' },
};
