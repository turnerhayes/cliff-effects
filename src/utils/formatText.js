import React from 'react';
import marksy from 'marksy';

const compile = marksy({ createElement: React.createElement });

const formatText = ({ text }) => {
  return compile(text).tree;
};

export default formatText;
