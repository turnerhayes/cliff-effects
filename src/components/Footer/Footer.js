import React from 'react';
import logo from '../../images/logo.svg';

import {
  Grid,
  Header,
  Segment,
  Image,
} from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';

import messages from './messages';

const Footer = () => {
  return (
    <Segment
      className = { `footer_segment` }
      inverted
      vertical
      color='teal'>
      <Grid
        container
        divided
        inverted
        stackable>
        <Grid.Row>
          <Grid.Column
            width={ 4 }
            floated='left'>
            <Header
              as='h4'
              inverted>
              <FormattedMessage
                { ...messages.header.text } />
            </Header>
            <p>
              <FormattedMessage
                { ...messages.cfbCredit } />
            </p>
          </Grid.Column>
          <a

            href="http://www.codeforboston.org"
            target="_blank"
            rel="noopener noreferrer"><Image
              src={ logo }
              size='small'
              floated='right' />
          </a>
        </Grid.Row>
      </Grid>
    </Segment>
  );
};

export default Footer;
