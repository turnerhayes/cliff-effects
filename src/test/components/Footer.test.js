import React from 'react';
import { mount } from 'enzyme';

import Footer from '../../components/Footer';
import { wrapWithIntlProvider } from '../utils/test-utils';

describe('<Footer>', () => {
  it('matches snapshot', () => {
    const wrapper = mount(
      wrapWithIntlProvider(
        <Footer />
      )
    );

    expect(wrapper.find(Footer)).toMatchSnapshot();
  });
});
