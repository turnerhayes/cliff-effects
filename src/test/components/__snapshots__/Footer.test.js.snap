// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> matches snapshot 1`] = `
<Footer>
  <Segment
    className="footer_segment"
    color="teal"
    inverted={true}
    vertical={true}
  >
    <div
      className="ui teal inverted vertical segment footer_segment"
    >
      <Grid
        container={true}
        divided={true}
        inverted={true}
        stackable={true}
      >
        <div
          className="ui container inverted stackable divided grid"
        >
          <GridRow>
            <div
              className="row"
            >
              <GridColumn
                floated="left"
                width={4}
              >
                <div
                  className="left floated four wide column"
                >
                  <Header
                    as="h4"
                    inverted={true}
                  >
                    <h4
                      className="ui inverted header"
                    >
                      <FormattedMessage
                        id="footer.header.text"
                        values={Object {}}
                      >
                        <span>
                          Cliff Effects Tool
                        </span>
                      </FormattedMessage>
                    </h4>
                  </Header>
                  <p>
                    <FormattedMessage
                      id="footer.cfbCredit"
                      values={Object {}}
                    >
                      <span>
                        Made with â™¥ by Code for Boston
                      </span>
                    </FormattedMessage>
                  </p>
                </div>
              </GridColumn>
              <a
                href="http://www.codeforboston.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <Image
                  as="img"
                  floated="right"
                  size="small"
                  src="logo.svg"
                  ui={true}
                >
                  <img
                    className="ui small right floated image"
                    src="logo.svg"
                  />
                </Image>
              </a>
            </div>
          </GridRow>
        </div>
      </Grid>
    </div>
  </Segment>
</Footer>
`;
